persist/sqlite/v2transactions.go

```diff
@@ -371,6 +352,70 @@ ORDER BY ts.transaction_order ASC`)
 	return nil
 }
 
+// fillV2TransactionFileContractRevisions fills in the file contracts for each
+// transaction.
+func fillV2TransactionFileContractRevisions(tx *txn, dbIDs []int64, txns []explorer.V2Transaction) error {
+	parentStmt, err := tx.Prepare(`SELECT fc.transaction_id, rev.confirmation_index, rev.confirmation_transaction_id, rev.resolution, rev.resolution_index, rev.resolution_transaction_id, fc.contract_id, fc.leaf_index, fc.capacity, fc.filesize, fc.file_merkle_root, fc.proof_height, fc.expiration_height, fc.renter_output_address, fc.renter_output_value, fc.host_output_address, fc.host_output_value, fc.missed_host_value, fc.total_collateral, fc.renter_public_key, fc.host_public_key, fc.revision_number, fc.renter_signature, fc.host_signature
+FROM v2_file_contract_elements fc
+INNER JOIN v2_transaction_file_contract_revisions ts ON (ts.parent_contract_id = fc.id)
+INNER JOIN v2_last_contract_revision rev ON (rev.contract_id = fc.contract_id)
+WHERE ts.transaction_id = ?
+ORDER BY ts.transaction_order ASC`)
+	if err != nil {
+		return fmt.Errorf("failed to prepare file contracts parent statement: %w", err)
+	}
+	defer parentStmt.Close()
+
+	revisionStmt, err := tx.Prepare(`SELECT fc.transaction_id, rev.confirmation_index, rev.confirmation_transaction_id, rev.resolution, rev.resolution_index, rev.resolution_transaction_id, fc.contract_id, fc.leaf_index, fc.capacity, fc.filesize, fc.file_merkle_root, fc.proof_height, fc.expiration_height, fc.renter_output_address, fc.renter_output_value, fc.host_output_address, fc.host_output_value, fc.missed_host_value, fc.total_collateral, fc.renter_public_key, fc.host_public_key, fc.revision_number, fc.renter_signature, fc.host_signature
+FROM v2_file_contract_elements fc
+INNER JOIN v2_transaction_file_contract_revisions ts ON (ts.revision_contract_id = fc.id)
+INNER JOIN v2_last_contract_revision rev ON (rev.contract_id = fc.contract_id)
+WHERE ts.transaction_id = ?
+ORDER BY ts.transaction_order ASC`)
+	if err != nil {
+		return fmt.Errorf("failed to prepare file contracts revision statement: %w", err)
+	}
+	defer revisionStmt.Close()
+
+	decorateTransactions := func(stmt *stmt, dbID int64, txnIndex int, revision bool) error {
+		rows, err := stmt.Query(dbID)
+		if err != nil {
+			return fmt.Errorf("failed to query file contracts: %w", err)
+		}
+		defer rows.Close()
+
+		j := 0
+		for rows.Next() {
+			fce, err := scanV2FileContract(rows)
```

Comment:
(nit) if you collect the `fce`s in the helper and then return them all at once instead of assigning them to the txn, you might be able to get rid of the `revision` and `txnIndex` args. Which might clean this up a bit more. Then you would just add 1 line per helper call in the loop which assigns the returned `fces`.